src/js/actions/export.js

```diff
@@ -180,23 +180,32 @@ define(function (require, exports) {
 
         return _exportService.exportAsset(document, layer, asset, fileName, baseDir)
             .bind(this)
-            .then(function (pathArray) {
-                // Do we need to be aware of exports that return >1 file path?
-                var assetProps = {
-                    filePath: pathArray[0],
-                    status: ExportAsset.STATUS.STABLE
-                };
-                return this.flux.actions.export.updateExportAsset(document, _layers, assetIndex, assetProps, true);
-            })
+            .then(
+                function (pathArray) {
+                    // Do we need to be aware of exports that return >1 file path?
+                    var assetProps = {
+                        filePath: pathArray[0],
+                        status: ExportAsset.STATUS.STABLE
+                    };
+                    return this.flux.actions.export.updateExportAsset(document, _layers, assetIndex, assetProps,
+                        true, true);
+                },
+                function (e) {
+                    log.error("Export Failed for asset %d of layerID %d, documentID %d, with error: ",
+                        assetIndex, layer && layer.id, document.id, e);
+
+                    var assetProps = {
+                        filePath: "",
+                        status: ExportAsset.STATUS.ERROR
+                    };
+
+                    return this.flux.actions.export.updateExportAsset(document, _layers, assetIndex, assetProps,
```

Comment:
Are you sure this is what you want to do here? I agree with the "enqueue an action" instead of "transfer to the action", but I don't think you want to block this action on the result of the "enqueue an action" promise. Almost always the two correct choices are 1) transfer and use the promise to block the parent action; or 2) enqueue a new action but ignore the promise.